src/sentry/api/endpoints/organization_events_trace.py

```diff
@@ -119,13 +128,58 @@ def serialize(self, result, root, snuba_event, event_id=None):
 
             # For the light response, the parent will be unknown unless it is a direct descendent of the root
             trace_results.append(
-                serialize_event(snuba_event, root["id"] if root_span is not None else None)
+                self.serialize_event(
+                    snuba_event, root["id"] if root_span is not None else None, None
+                )
             )
 
         event = eventstore.get_event_by_id(snuba_event["project_id"], event_id)
         for span in event.data.get("spans", []):
             if span["span_id"] in parent_map:
                 child_event = parent_map[span["span_id"]]
-                trace_results.append(serialize_event(child_event, event_id))
+                trace_results.append(self.serialize_event(child_event, event_id, None))
 
         return trace_results
+
+
+class OrganizationEventsTraceEndpoint(OrganizationEventsTraceEndpointBase):
+    def serialize_event(self, *args, **kwargs):
+        event = super().serialize_event(*args, **kwargs)
+        event["children"] = []
+        return event
+
+    def serialize(self, parent_map, root, warning_extra, snuba_event=None, event_id=None):
+        """ For the full event trace, we return the results as a graph instead of a flattened list """
+        parent_events = {}
+        parent_events[root["id"]] = result = self.serialize_event(root, None, 0, True)
```

Comment:
This is mostly me nitpicking here but can we either break the assignment into two lines here or swap the order to be `result = parent_events[root["id"]] = ...`? Almost didn't see `result` here.

Actually, it looks like `result` isn't even needed here? You can just return `parent_events[root["id']]` instead.

Reply:
Gonna stick with just `result` cause I think it's easier to read, but I'll flip the assignment order :+1: